integration-tests/tests/src/tests/queries.ts

```diff
@@ -82,6 +92,43 @@ class Story extends Realm.Object<Story> implements IStory {
   };
 }
 
+class MyGeoPoint implements IGeoPoint {
```

Comment:
Just an idea, but I think it would be better if we provided classes that the developer could import and use in their schema, such as GeoPoint, GeoCircle, etc.  The structure of these embedded classes are very specific and could possibly be messed up (ie the `type` string).

Then the developer could just import the class, add it to their `schema` config and use the geo type in other schemas.

Reply:
The interface is a good start, since it will show a type error if `type` is not set to the correct string.  I hadn't read the last part of the tech design, but now I see that the types will be coming in core.  Hopefully the migration from this to having an internal type is not a large lift for our users.